config/auth/authconfig.go

```diff
@@ -25,13 +25,13 @@ import (
 
 // Config is the CLI's authentication configuration.
 type Config struct {
-	Enabled          bool           `yaml:"enabled"`
-	IgnoreCertErrors bool           `yaml:"ignoreCertErrors"`
-	X509             *x509.Config   `yaml:"x509,omitempty"`
-	OAuth2           *oauth2.Config `yaml:"oauth2,omitempty"`
-	Basic            *basic.Config  `yaml:"basic,omitempty"`
-	Iap              *config.Config `yaml:"iap,omitempty"`
-	Ldap             *ldap.Config   `yaml:"ldap,omitempty"`
+	Enabled          bool           `json:"enabled" yaml:"enabled"`
+	IgnoreCertErrors bool           `json:"ignoreCertErrors" yaml:"ignoreCertErrors"`
+	X509             *x509.Config   `json:"x509,omitempty" yaml:"x509,omitempty"`
+	OAuth2           *oauth2.Config `json:"oauth2,omitempty" yaml:"oauth2,omitempty"`
+	Basic            *basic.Config  `json:"basic,omitempty" yaml:"basic,omitempty"`
+	Iap              *config.Config `json:"iap,omitempty" yaml:"iap,omitempty"`
+	Ldap             *ldap.Config   `json:"ldap,omitempty" yaml:"ldap,omitempty"`
 
-	GoogleServiceAccount *gsa.Config `yaml:"google_service_account,omitempty"`
+	GoogleServiceAccount *gsa.Config `json:"google_service_account,omitempty" yaml:"google_service_account,omitempty"`
```

Comment:
note: this change requires manual patching of pre-existing ~/.spin/config file as now `GoogleServiceAccount: null` doesn't parse